cloudmarker/events/azvmosdiskencryptionevent.py

```diff
@@ -0,0 +1,110 @@
+"""Microsoft Azure VM OS disk encryption event.
+
+This module defines the :class:`AzVMOSDiskEncryptionEvent` class that
+identifies an unencrypted Azure OS disk. This plugin works on the
+virtual machine properties found in the ``com`` bucket of
+``virtual_machine`` records.
+"""
+
+
+import logging
+
+from cloudmarker import util
+
+_log = logging.getLogger(__name__)
+
+
+class AzVMOSDiskEncryptionEvent:
+    """Az VM OS disk encryption event plugin."""
+
+    def __init__(self):
+        """Create an instance of :class:`AzVMOSDiskEncryptionEvent`."""
+
+    def eval(self, record):
+        """Evaluate Azure virtual machine to check for unencrypted OS disk.
+
+        Arguments:
+            record (dict): A virtual machine record.
+
+        Yields:
+            dict: An event record representing an unencrypted OS disk
+            of an Azure virtual machine
+
+        """
+        # If 'com' bucket is missing, we have a malformed record. Log a
+        # warning and ignore it.
+        com = record.get('com')
+        if com is None:
+            _log.warning('Virtual machine record is missing \'com\' key: %r',
+                         record)
+            return
+
+        # This plugin understands compute rule records only,
+        # so ignore  any other record types.
+        common_record_type = com.get('record_type')
+
+        if common_record_type != 'compute':
+            return
+
+        # If 'ext' bucket is missing, we have a malformed record. Log a
+        # warning and ignore it.
+        ext = record.get('ext')
```

Comment:
Can we put these checks in a single method which logs warning in case a specific bucket is missing.